decidim/decidim
# Fix for endorsed_by with other user group's member

#### :tophat: What? Why?

When a user group has endorsed a resource (for instance, a proposal) and another member of the user group with permissions go to unendorse it, they cannot do that. 

This PR fixes that. 

I've also improved the specs by moving the endorsable examples to a shared example for better reusing without copying and pasting. 

#### :pushpin: Related Issues

- Related to #5406 (I found this bug while checking that out to see if it was already fixed) 

#### Testing

1. Add two participants to the same user group with "admin" role 
2. Go to a proposal, endorse it as user group with one user
3. Open another browser, sign in with other user, go to endorse the same proposal
4. See that your user group isn't selected, and you can't unendorse it 
 
### :camera: Screenshots

#### Before

![image](https://user-images.githubusercontent.com/717367/156738235-46a606a0-fea3-40e1-bcfd-3db5bc8b5216.png)


#### After

![image](https://user-images.githubusercontent.com/717367/156737898-005ef3d2-f301-430c-a27e-8a7facee1e4c.png)


:hearts: Thank you!
